src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -701,6 +714,28 @@ JNI_METHOD(void, pairDevice)
     }
 }
 
+JNI_METHOD(void, pairDeviceThroughNfc)
+(JNIEnv * env, jobject self, jlong handle, jlong deviceId, jlong pinCode, jbyteArray csrNonce, jobject networkCredentials,
+ jobject icdRegistrationInfo)
+{
+    chip::DeviceLayer::StackLock lock;
+    AndroidDeviceControllerWrapper * wrapper = AndroidDeviceControllerWrapper::FromJNIHandle(handle);
+
+    ChipLogProgress(Controller, "pairDeviceThroughNfc() called with device ID and pincode");
+
+    if (!chip::CanCastTo<uint32_t>(pinCode))
+    {
+        JniReferences::GetInstance().ThrowError(env, sChipDeviceControllerExceptionCls, CHIP_ERROR_INVALID_ARGUMENT);
+        return;
+    }
+
+    RendezvousParameters rendezvousParams =
+        RendezvousParameters().SetSetupPINCode(static_cast<uint32_t>(pinCode)).SetPeerAddress(Transport::PeerAddress::NFC());
```

Comment:
Looking at things, the only difference between `pairDeviceThroughNfc` and `pairDevice` seems to be the way the rendezvousparams are set ... this is NFC where as the other one is BLE (and pin code is always the same).

In this case it seems to make more sense to make that whole part common/shared and called pair device and not copy & paste the prefix and share suffix only.

I would suggest the following changes:

  - JNI rename the pairDevice to `pairDeviceThroughBLE`
  - add this `pairDeviceThrougNfc` method as a new method
  - make both use the same `pariDevice` static method that takes the PeerAddress as an argument
  
  That would allow for more code reuse and more consistent naming.